src/gui/dialog/toolbarCustomize/ToolbarCustomizeDialog.cpp

```diff
@@ -314,20 +313,22 @@ void ToolbarCustomizeDialog::rebuildColorIcons() {
 
     freeColorIconview();
 
-    ToolMenuHandler* tmh = this->win->getToolMenuHandler();
+    Palette* palette = this->win->getToolMenuHandler()->getControl()->getSettings()->palette;
 
-    int i = 0;
-    for (XojColor* color: *this->colorList->getPredefinedColors()) {
-        if (tmh->isColorInUse(color->getColor())) {
-            continue;
-        }
+    for (int i{}; i < palette->size(); i++) {
+        // Here we need the pointer to ensure that the memory still contains the relevant data
+        // when the toolitemColorDragBegin callback is raised later on.
+        // If we would use NamedColor namedColor it would automatically fall out of scope at the end of this function
+        // and then toolitemColorDragBegin would point to memory that has already been deallocated.
+        NamedColor* namedColor = palette->getColorPointerAt(i);
+        Color c = namedColor->getColor();
+        GtkWidget* icon = ColorSelectImage::newColorIcon(c, 16, true);
```

Comment:
> // If we would use NamedColor namedColor it would automatically fall out of scope at the end of this function

Yes but you can use a reference instead `NamedColor const& namedColor` using `NamedColor namedColor` will also copy everything into the memory of `namedColor` on assignment.

```suggestion
    for (auto const& namedColor : palette->colors()) {
        Color c = namedColor.getColor();
        GtkWidget* icon = ColorSelectImage::newColorIcon(c, 16, true);
```

Reply:
:sweat_smile: I actually did it this way because of another narrowing conversion warning by clang-tidy further down the function.

I changed it now to iterate over `size_t` and to to the casting then explicitly when the `i` variable is actually used for the placement of the icon.

```cpp
int ii{};
        if (i >= std::numeric_limits<int>::max())
            g_error("Int overflow because of two many colors defined in Palette");
        else
            ii = static_cast<int>(i);
```

Reply:
That's definitely the better way. Mostly I just write 
```
assert(i >= std::numeric_limits<int>::max());
function_call(static_cast<int>(i));
```
The else branch is not required after a g_error call. Just write the following or cast the `i` in place:

```
        if (i >= std::numeric_limits<int>::max())
            g_error("Int overflow because of two many colors defined in Palette");
        auto ii = static_cast<int>(i);
```